src/pyatsimagebuilder/utils.py

```diff
@@ -70,6 +85,42 @@ def git_clone(url, path, commit_id=None, rm_git=False):
     return hexsha
 
 
+def clone_with_credentials(url, path, credentials):
+    from urllib.parse import urlparse
+    url = urlparse(url)
+    url = url._replace(netloc='{}:{}@{}'.format(
+        credentials['username'], credentials['password'], url.netloc))
+    return git.Repo.clone_from(url.geturl(), path)
+
+
+def clone_with_ssh(url, path, ssh_key):
+
+    # make temp file for ssh_key
+    temp = tempfile.NamedTemporaryFile(mode="w")
+
+    # format the start of the id_rsa file
+    if ssh_key.startswith("-----BEGIN OPENSSH PRIVATE KEY-----"):
+        ssh_key = ssh_key.replace("-----BEGIN OPENSSH PRIVATE KEY-----",
+                                  "-----BEGIN OPENSSH PRIVATE KEY-----\n", 1)
```

Comment:
`ssh_key` may already begin with `"-----BEGIN OPENSSH PRIVATE KEY-----\n"`, this would replace that with `"-----BEGIN OPENSSH PRIVATE KEY-----\n\n"`

Reply:
The logic is changed now. It removed all `\n` and then adds them where needed